src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3272,8 +3272,8 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         resource = cmd.cli_ctx.cloud.endpoints.active_directory_resource_id
         cred, _, _ = profile.get_login_credentials(resource=resource,
                                                    aux_subscriptions=aux_subscriptions)
-        _, _, _, external_tokens = cred.get_all_tokens('https://management.azure.com/.default')
-        external_bearer_token = external_tokens[0][0] + ' ' + external_tokens[0][1]
+        _, external_tokens = cred.get_all_tokens('https://management.azure.com/.default')
+        external_bearer_token = 'Bearer' + ' ' + external_tokens[0].token
```

Comment:
Changed the code to use [`azure.core.credentials.AccessToken`](https://github.com/Azure/azure-sdk-for-python/blob/c69590242bcca2a0f60521e2846b68b9ac2a9640/sdk/core/azure-core/azure/core/credentials.py#L31):

```py
AccessToken = namedtuple("AccessToken", ["token", "expires_on"])
```

instead of the `(tokenType, accessToken, token_entry)` tuple (`tokenType` is always `'Bearer'`) returned by functions like `retrieve_token_for_user`:

https://github.com/Azure/azure-cli/blob/2d27b9fd1c36603a5fdc9614cea75c9e82637293/src/azure-cli-core/azure/cli/core/_profile.py#L1072

Reply:
@qwordy

Reply:
OK